.github/workflows/tests.yml

```diff
@@ -0,0 +1,78 @@
+# This workflow builds and runs test
+
+name: tests
+
+on:
+  pull_request:
+    branches:
+      - "main"
+      - "develop"
+    paths-ignore:
+      - "docs/**"
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        py-version:
+          - 3.6
+          - 3.7
+          - 3.8
+        mpi:
+          - openmpi
+        os:
+          - ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.head_ref }}
+
+      - name: Install MPI (${{ matrix.mpi }})
+        run: bash .bash/install-${{ matrix.mpi }}.sh
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+
+      - name: Install via pip
+        run: |
+          sudo pip install -e .[all]
+
+      - name: Run serial tests
+        run: |
+          pytest -v
+
+      - name: Run parallel tests (procs=2)
+        run: |
+          mpirun -n 2 --oversubscribe python -m pytest test/parallel_tests/ --with-mpi
+
+      - name: Run parallel tests (procs=4)
+        run: |
+          mpirun -n 4 --oversubscribe python -m pytest test/parallel_tests/ --with-mpi
+
+      - name: Run parallel tests (procs=7)
+        run: |
+          mpirun -n 7 --oversubscribe python -m pytest test/parallel_tests/ --with-mpi
+
+      - name: Run parallel tests (procs=12)
+        run: |
+          mpirun -n 12 --oversubscribe python -m pytest test/parallel_tests/ --with-mpi
+
+      - name: Run tests with coverage
+        run: |
```

Comment:
This should happen only when merging with the master, since the badge is updated globally.
I will take care of fixing this!